freeCodeCamp/fcctesting.js

```diff
@@ -37,44 +37,62 @@
 
 const fs = require('fs');
 
-const allowedOriginsMatcher = /^https?:\/\/([\w-]+\.)*freecodecamp.org/;
+const allowedOriginsMatcher = process.env.NODE_ENV === 'production' ? /^https?:\/\/([\w-]+\.)*freecodecamp.org/ : /^http:\/\/localhost:8000/;
 
 module.exports = function (app) {
-  
+
   app.use(function (req, res, next) {
-      const origin = req.get('origin');
-      if (allowedOriginsMatcher.test(origin)) {
-            res.setHeader('Access-Control-Allow-Origin', origin);
-      }
-      res.setHeader('Access-Control-Allow-Credentials', true);
-      next();
+    const origin = req.get('origin');
+    if (allowedOriginsMatcher.test(origin)) {
+      res.setHeader('Access-Control-Allow-Origin', origin);
+    }
+    res.setHeader('Access-Control-Allow-Credentials', true);
+    next();
   });
```

Comment:
Can we simplify this? I ran into some problems with `Access-Control-Allow-Origin` and setting the environment variable in the `start` script while working through this on Repl.it.

This worked on my end, though I'm sure there's another way to do it:

```js
const allowedOrigins = [/^https?:\/\/([\w-]+\.)*freecodecamp.org/, /^http:\/\/localhost:\d+/];

module.exports = function (app) {

  app.use(function (req, res, next) {
    const origin = req.get('origin');
    if (allowedOrigins.some(regex => regex.test(origin))) {
      res.setHeader('Access-Control-Allow-Origin', origin);
      console.log(origin);
    }

    res.setHeader('Access-Control-Allow-Credentials', true);
    next();
  });
```

Reply:
Ah...yes. I do not know why I made it so complicated with the env variable. Also, here is a link to a working project: https://repl.it/@Sky020/boilerplate-advancednode#index.js

It is off this branch: https://github.com/Sky020/boilerplate-advancednode/tree/merged-master-boilerplates

I had to change `server.js` to `index.js`, because I cannot find a way to tell repl.it to use anything other than `index.js` as an entry point. So, if you know how to do this, that would be useful.

Reply:
@Sky020, no worries. Thanks for all of your hard work on the boilerplate and all those challenges.

This worked locally, but cors and all those headers are really tough. That's what worked for me in the end, but I'm sure there's a better way to go about it.

About `server.js` or `index.js` on repl.it, I didn't have a problem with either in your project or mine here: https://repl.it/@scissorsneedfoo/boilerplate-advancednode-1

How were you running the project @Sky020? Did you click the "Run" button, or did you use console commands?

It's interesting, but when I forked your project and copied everything from `index.js` into `server.js` I wasn't able to go back and delete `index.js` for some reason. The only options I got were "Open tab" and "Copy link". Do you know what might be going on here @Obaida-Albaroudi?

Reply:
>How were you running the project @Sky020?

I did very little, at first. The window asking me to configure how to run the project popped up, and I left it at default with `npm start`. Then, I used the _"Run"_ button, and the console spat out an error about not being able to find `index.js`. So, I renamed `server.js` to `index.js` and it worked.

Reply:
FYI - Repl.it expects index.js to be the entry point.  You can not change it.

Reply:
Good to know, thanks @RandellDawson.

It's funny, but all of our recently ported projects use `server.js` rather than `index.js`. I'll double check with the Repl.it team to make sure that won't be a problem.